src/SFML/Graphics/Image.cpp

```diff
@@ -131,6 +134,27 @@ sfBool sfImage_saveToFile(const sfImage* image, const char* filename)
 }
 
 
+////////////////////////////////////////////////////////////
+sfBool sfImage_saveToMemory(const sfImage* image, unsigned char** buffer, size_t* size, const char* format)
+{
+    CSFML_CHECK_RETURN(image, sfFalse);
+
+    std::vector<sf::Uint8> output;
+    if (image->This.saveToMemory(output, format)){
+        // Create new buffer then return address and size to user
+        *buffer = reinterpret_cast<unsigned char*>(std::malloc(output.size()));
```

Comment:
Wonder if we could/should assume the buffer to already be properly prepared. 🤔 

Like from a user perspective, I wonder whether one would really pass in some raw pointers and expect the function to allocate space for it.

Reply:
I believe the usage looks like this:
```c
unsigned char* buffer;
size_t size;
sfImage_saveToFile(image, &buffer, &size, "someformatstring");
```
Given C's limitations, this was the best API I could come up with. The problem with requiring users to prepare the buffer is that I'm not sure how they're supposed to correctly size that buffer. What I wrote means users don't have to know or care what the size it beforehand.

Reply:
I think it's fine to malloc here. Requiring the user the do a malloc would in a sense, remove the "Simple" in SFML IMHO.

Reply:
It's impossible for the user to predict how big the buffer should be. After all, image compression makes it so the data size can be anything whatsoever. I think this solution as suggested is the best that can be done with the given API.

Reply:
But it is true that this is the first ever usage of `malloc` in this repository (previously it was all `new`). And it's possible that the user is using yet another kind of allocator but will need to deal with `free` specifically only for this use case.